packages/cosmos/src/gas_price.rs

```diff
@@ -93,10 +93,10 @@ impl GasPriceMethod {
                     }
                 };
                 if should_trigger {
-                    let client = client.clone();
+                    let builder = builder.clone();
```

Comment:
Slightly concerned with this approach for two reasons:

1. Minor point, but we're cloning a much larger data structure than we need
2. The original builder may end up with different settings after this is passed in

After reading the rest of the code, I think the core problem isn't here, it's the fact that we're trying to create a new `Cosmos` inside the update loop.

Reply:
does this address your concerns here (and https://github.com/Levana-Protocol/levana-cosmos-rs/pull/69#discussion_r1458355134)? I added a more descriptive comment in the code too: https://github.com/Levana-Protocol/levana-cosmos-rs/pull/69/commits/9f3a3403ab5fdbfbfc5e40ff521e6c0314bb0c2d

Reply:
Can we always use a default gas price until the builder has completed, and only then switch to the dynamic price?

Reply:
the switching would still be contained in the library code, I'm thinking inside of `build_lazy()` before returning the Cosmos instance

Reply:
Yes, that's fine.

Reply:
refactored in https://github.com/Levana-Protocol/levana-cosmos-rs/pull/69/commits/219fbb03791e62707d60ce0493d71299e176fe7e

Reply:
actually wait, I think I can get rid of the lock... one min

Reply:
nah, not worth it, leaving as-is for now (irrelevant details: I thought we could have different variants in the gas price method, like a state machine, and move it from Default to Eip when Cosmos is ready - but ran into trouble trying to replace it in build_lazy since we'd need builder to be mutable, and can't hold onto that beyond creating the Cosmos instance)